src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -106,13 +113,18 @@ public final class TestResult extends MetaTabulatedResult {
 
     private float duration;
 
+    private boolean parseOldReports = true;
```

Comment:
Again will not be read from disk as such.

Could this field be `transient`? I imagine `parse` is only called from the same Jenkins session as the `TestResult` constructor? I think it does need to be a field rather than a local variable, unfortunately, due to https://github.com/jenkinsci/junit-plugin/blob/83cdce2b26ab579da34b344063b599bfe7985c6f/src/main/java/hudson/tasks/junit/JUnitResultArchiver.java#L256-L258 if I read it right.